### PR TITLE
feat: Add styleing for darkmode

### PR DESCRIPTION
**Pull request by littletof**
Add darkmode support for the index page (the ignore rules page was handled already).

The diff on github looks worst than what actually changed.

<details>
<summary>Previous</summary>

image

</details>

Now:

image

### CODE REVIEW

**Comment by littletof** [comment]
Not fully sold on the searchbar styleing, but it works as a start anyways.

**Comment by littletof** [comment]
@hashrock Not urgent, but can you please review this PR and #1066?
These should fix darkmode and markdown styleing together with the #1067.

**Comment by bartlomieju** [comment]
Opened https://github.com/denoland/deno_lint/pull/1069

File: www/pages/index.tsx [hunk 1]
```diff
@@ -39,53 +39,55 @@ function IndexPage(props: PageProps) {
...
+            </div>
+          </form>
+          <script
+            dangerouslySetInnerHTML={{
+              __html:
+                "document.getElementById('all_rules').oninput = () => document.getElementById('search_form').submit();",
+            }}
+          >
+          </script>
+          <div class={tw`mt-6 text-gray-600`}>
```
Comment (hashrock) [inline www/pages/index.tsx:82]:
```suggestion
          <div class={tw`mt-6 text-gray-600 dark:text-gray-400`}>
```

Not enough contrast here:
image

**Review by hashrock — CHANGES_REQUESTED** [review]
very clean darkmode theme!
I found a small accessibility issue here

**Comment by littletof** [comment]
@hashrock Right, thanks, fixed it with your suggestion. Looks better now.

**Comment by littletof** [comment]
@hashrock @bartlomieju can this be merged?

**Pull request merged**